packages/@apphosting/adapter-nextjs/src/bin/build.ts

```diff
@@ -4,16 +4,27 @@ import {
   build,
   populateOutputBundleOptions,
   generateOutputDirectory,
+  DEFAULT_COMMAND,
 } from "../utils.js";
-
 import { join } from "path";
 
-const cwd = process.cwd();
+const root = process.cwd();
+
+let projectRoot = root;
+if (process.env.MONOREPO_PROJECT && process.env.FIREBASE_APP_DIRECTORY) {
```

Comment:
OOC why do we need the MONOREPO_PROJECT check? Shouldn't we always respect FIREBASE_APP_DIRECTORY?

Reply:
Yes we should always be respecting FIREBASE_APP_DIRECTORY.